jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -947,17 +947,9 @@ def _fit_model_to_cube(self, add_data):
                 temp_label = "{} ({}, {})".format(self.results_label, m["x"], m["y"])
                 self.app.fitted_models[temp_label] = m["model"]
 
-        count = max(map(lambda s: int(next(iter(re.findall(r"\d$", s)), 0)),
-                        self.data_collection.labels)) + 1
-
-        label = self.app.return_data_label(f"{self.results_label}[Cube]", ext=count)
-
-        # Create new glue data object
-        output_cube = Data(label=label,
-                           coords=fitted_spectrum.wcs,
-                           flux=fitted_spectrum.flux.value)
-        output_cube.get_component('flux').units = fitted_spectrum.flux.unit.to_string()
+        output_cube = Spectrum1D(flux=fitted_spectrum.flux, wcs=fitted_spectrum.wcs)
```

Comment:
I don't know if extra transposing is needed or not, so please double check whether your output cube is flipped w.r.t. input or not. I cannot tell from the test data Cami gave me because it has same dimension on both X and Y.